docs/concepts/ui-embedder-context-api.md

```diff
@@ -0,0 +1,60 @@
+# EmbedderContext API
```

Comment:
This is a proposal for the embedder support API discussed in an off-line meeting. Obviously, it works for my application but I welcome any suggestions for further improvement, especially integration points that I may have missed in parts of the Perfetto UI app that I'm not using.

Also, this PR includes changes to certain aspects of the build and development environment that support developing an application against the Perfetto APIs. That can be factored out into a separate PR if preferred.